tests/fixtures/optimizer/simplify.sql

```diff
@@ -911,13 +911,58 @@ t1.a = 39 AND t2.b = t1.a AND t3.c = t2.b;
 t1.a = 39 AND t2.b = 39 AND t3.c = 39;
 
 x = 1 AND CASE WHEN x = 5 THEN FALSE ELSE TRUE END;
-x = 1 AND CASE WHEN FALSE THEN FALSE ELSE TRUE END;
+x = 1;
 
 x = 1 AND IF(x = 5, FALSE, TRUE);
-x = 1 AND CASE WHEN FALSE THEN FALSE ELSE TRUE END;
+x = 1;
+
+-- Simplified to x = 1 if the canonicalize rule runs first
+x = 1 AND CASE x WHEN 5 THEN FALSE ELSE TRUE END;
+x = 1 AND CASE 1 WHEN 5 THEN FALSE ELSE TRUE END;
 
 x = y AND CASE WHEN x = 5 THEN FALSE ELSE TRUE END;
 x = y AND CASE WHEN x = 5 THEN FALSE ELSE TRUE END;
 
 x = 1 AND CASE WHEN y = 5 THEN x = z END;
 x = 1 AND CASE WHEN y = 5 THEN 1 = z END;
+
+--------------------------------------
+-- Simplify Conditionals
+--------------------------------------
+IF(TRUE, x, y);
+x;
+
+IF(FALSE, x, y);
+y;
+
+IF(FALSE, x);
+NULL;
+
+IF(NULL, x, y);
+CASE WHEN NULL THEN x ELSE y END;
```

Comment:
this seems suboptimal, this should always be y

Reply:
Most of the dialects I mentioned which support `null` as a condition value treat it as falsey, so maybe this could be done. I'm curious though about why we have `always_true` and we don't have `always_false` which takes nulls into account?

Reply:
always_false should take nulls into account

Reply:
Done https://github.com/tobymao/sqlglot/pull/2403/commits/a5ad243e0dba7434243bac0d8488be0fb801ed39